migrations/2024-08-28-124527_add-repo-submissions-table/up.sql

```diff
@@ -0,0 +1,25 @@
+-- Your SQL goes here
+CREATE TABLE repositories (
+    id UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+    user_id UUID NOT NULL REFERENCES users(id),
+    challenge_id UUID NOT NULL REFERENCES challenges(id),
+    repo_url VARCHAR(255) NOT NULL,
+    created_at TIMESTAMP NOT NULL DEFAULT now(),
+    updated_at TIMESTAMP NOT NULL DEFAULT now()
+);
+
+CREATE TABLE submissions (
+    id UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+    user_id UUID NOT NULL REFERENCES users(id),
+    exercise_id UUID NOT NULL REFERENCES exercises(id),
+    commit_id VARCHAR(255) NOT NULL,
+    repository_id UUID NOT NULL REFERENCES repositories(id),
+    status VARCHAR(255) NOT NULL CHECK (status IN ('pending', 'passed', 'failed')),
+    feedback TEXT,
+    submitted_at TIMESTAMP NOT NULL DEFAULT now(),
+    updated_at TIMESTAMP NOT NULL DEFAULT now()
```

Comment:
This will be set to datetime at time of row creation and will need to be updated manually on every update after that (unless we add a database trigger).  Just something to note.

Reply:
Diesel automatically added a trigger SQL script that automatically updated the `updated_at` field whenever there's an update to the record which was why I did add the trigger in the migration scripts. see here -> https://github.com/Extheoisah/hxckr-core/pull/10/files#diff-79ed83666bddb98b5ca0f9f9f009f0208d86e218a232fea1f4f6dc503b2b9b5bR19

Reply:
I tested that this trigger works. See timestamp in attached screenshot
<img width="215" alt="Screenshot 2024-08-30 at 14 35 56" src="https://github.com/user-attachments/assets/276356dd-d510-4deb-8769-eb555e6feae2">

Reply:
I think you need to use the trigger function and reference the table name for it to work?  I could be wrong.  There's an example in the comments above that function you linked to.

Reply:
Actually, during my testing, I didn't include it anywhere. I just updated the leaderboard record and the updated at field was automatically updated. I see from my db inspector that the trigger function is already added automatically by diesel
<img width="1005" alt="Screenshot 2024-08-30 at 14 39 35" src="https://github.com/user-attachments/assets/60f2fd14-df28-4292-9c6f-017f98c8d16f">

Reply:
So I cross-checked again to be sure that I was correct, but I was wrong, and you're right. The function wasn't automatically applied, and the reason I saw that the leaderboard record was updated was that I was manually updating the updated_at field in the update function.
I have added a migration script to reference the diesel_manage_updated_at trigger function for all tables except for the session table. 
Sorry for the mix-up. See changes here -> https://github.com/Extheoisah/hxckr-core/pull/10/commits/c2f5744b467abe37a75c031bedbf54a2939c46ca